File: easy/58. Length of Last Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        length = 0
        i = len(s) - 1
        while i >= 0 and s[i] == ' ': # solve the ' '
            i -= 1
        while i >= 0 and s[i] != ' ': # cal length
            length += 1
            i -= 1
        return length

File: easy/58. Length of Last Word.cpp
#include <string>

class Solution {
public:
    int lengthOfLastWord(string s) {
        int length = 0;
        int i = s.size() - 1;
        
        // Skip trailing spaces
        while (i >= 0 && s[i] == ' ') {
            i--;
        }
        
        while (i >= 0 && s[i] != ' ') {
            length++;
            i--;
        }
        return length;
    }
};